lib/helpers/mail/mail.js

```diff
@@ -2,928 +2,48 @@
-      this.contents = [];
-    }
-    this.contents.push(content);
-  };
-
-  this.getContents = function() {
-    return this.contents;
-  };
-
-  this.addAttachment = function(attachment) {
-    if (this.attachments === undefined) {
-      this.attachments = [];
-    }
-    this.attachments.push(attachment);
-  };
-
-  this.getAttachments = function() {
-    return this.attachments;
-  };
-
-  this.setTemplateId = function(template_id) {
-    this.template_id = template_id;
-  };
-
-  this.getTemplateId = function() {
-    return this.template_id;
-  };
-
-  this.addSection = function(section) {
-    if (this.sections === undefined) {
-      this.sections = {};
-    }
-    this.sections[Object.keys(section)[0]] = section[Object.keys(section)[0]];
-  };
-
-  this.getSections = function() {
-    return this.sections;
-  };
-
-  this.addHeader = function(header) {
-    if (this.headers === undefined) {
-      this.headers = {};
-    }
-    this.headers[Object.keys(header)[0]] = header[Object.keys(header)[0]];
-  };
+  this.from = {};
+  this.personalizations = [];
+  this.subject = '';
+  this.content = [];
 
-  this.getHeaders = function() {
-    return this.headers;
-  };
-
-  this.addCategory = function(category) {
-    if (this.categories === undefined) {
-      this.categories = [];
-    }
-    this.categories.push(category.category);
-  };
-
-  this.getCategories = function() {
-    return this.categories;
-  };
-
-  this.addCustomArg = function(custom_arg) {
-    if (this.custom_args === undefined) {
-      this.custom_args = {};
-    }
-    this.custom_args[Object.keys(custom_arg)[0]] =
-      custom_arg[Object.keys(custom_arg)[0]];
-  };
-
-  this.getCustomArgs = function() {
-    return this.custom_args;
-  };
-
-  this.setSendAt = function(send_at) {
-    this.send_at = send_at;
-  };
-
-  this.getSendAt = function() {
-    return this.send_at;
-  };
-
-  this.setBatchId = function(batch_id) {
-    this.batch_id = batch_id;
-  };
-
-  this.getBatchId = function() {
-    return this.batch_id;
-  };
-
-  this.setAsm = function(asm) {
-    this.asm = asm;
-  };
-
-  this.getAsm = function() {
-    return this.asm;
-  };
-
-  this.setIpPoolName = function(ip_pool_name) {
-    this.ip_pool_name = ip_pool_name;
-  };
-
-  this.getIpPoolName = function() {
-    return this.ip_pool_name;
-  };
-
-  this.addMailSettings = function(mail_settings) {
-    this.mail_settings = mail_settings;
-  };
-
-  this.getMailSettings = function() {
-    return this.mail_settings;
-  };
-
-  this.addTrackingSettings = function(tracking_settings) {
-    this.tracking_settings = tracking_settings;
-  };
-
-  this.getTrackingSettings = function() {
-    return this.tracking_settings;
-  };
+  this.addFromEmail(from_email);
+  this.addSubject(subject);
+  this.addToEmail(to_email);
+  this.addContent(content);
+}
 
-  this.setReplyTo = function(reply_to) {
-    this.reply_to = reply_to;
-  };
+Mail.prototype.addFromEmail = function (email) {
```

Comment:
Just a suggestion, I think it would be good to stick with the previous convention of `setX` method signature for properties that are not arrays, and `addX` for arrays.